### ISSUE TITLE
Error in sklearn$metrics$precision_score(y_test, y_pred, average = "weighted") :    attempt to apply non-function

### ISSUE THREAD

**Issue by ZijiaLai** [dzhakparov/GeneSelectR#30]
I encountered an error while trying to use the `precision_score` function from `sklearn.metrics` in an R environment using the `reticulate` package. The error is:
`Error in sklearn$metrics$precision_score(y_test, y_pred, average = "weighted") : 
  attempt to apply non-function`

`> selection_results <- GeneSelectR(X = X, y = y,njobs = -1,
+                                  calculate_permutation_importance=TRUE,
+                                  perform_test_split=TRUE) # all cores will be used used
Performing feature selection procedure. Please wait, it takes some time
$preprocessing
Module(sklearn.preprocessing)

$model_selection
Module(sklearn.model_selection)

$feature_selection
Module(sklearn.feature_selection)

$ensemble
Module(sklearn.ensemble)

$pipeline
<class 'sklearn.pipeline.Pipeline'>
 signature: (steps, memory=None, verbose=False)

$forest
<class 'sklearn.ensemble._forest.RandomForestClassifier'>
 signature: (
   n_estimators=100, 
   criterion='gini', 
   max_depth=None, 
   min_samples_split=2, 
   min_samples_leaf=1, 
   min_weight_fraction_leaf=0.0, 
   max_features='auto', 
   max_leaf_nodes=None, 
   min_impurity_decrease=0.0, 
   min_impurity_split=None, 
   bootstrap=True, 
   oob_score=False, 
   n_jobs=None, 
   random_state=None, 
   verbose=0, 
   warm_start=False, 
   class_weight=None, 
   ccp_alpha=0.0, 
   max_samples=None
)

$boruta
Module(boruta)

$randomized_grid
<class 'sklearn.model_selection._search.RandomizedSearchCV'>
 signature: (
   estimator, 
   param_distributions, 
   n_iter=10, 
   scoring=None, 
   n_jobs=None, 
   iid='deprecated', 
   refit=True, 
   cv=None, 
   verbose=0, 
   pre_dispatch='2*n_jobs', 
   random_state=None, 
   error_score=nan, 
   return_train_score=False
)

$grid
<class 'sklearn.model_selection._search.GridSearchCV'>
 signature: (
   estimator, 
   param_grid, 
   scoring=None, 
   n_jobs=None, 
   iid='deprecated', 
   refit=True, 
   cv=None, 
   verbose=0, 
   pre_dispatch='2*n_jobs', 
   error_score=nan, 
   return_train_score=False
)

$bayesianCV
<class 'skopt.searchcv.BayesSearchCV'>
 signature: (
   estimator, 
   search_spaces, 
   optimizer_kwargs=None, 
   n_iter=50, 
   scoring=None, 
   fit_params=None, 
   n_jobs=1, 
   n_points=1, 
   iid='deprecated', 
   refit=True, 
   cv=None, 
   verbose=0, 
   pre_dispatch='2*n_jobs', 
   random_state=None, 
   error_score='raise', 
   return_train_score=False
)

$lasso
<class 'sklearn.linear_model._logistic.LogisticRegression'>
 signature: (
   penalty='l2', 
   dual=False, 
   tol=0.0001, 
   C=1.0, 
   fit_intercept=True, 
   intercept_scaling=1, 
   class_weight=None, 
   random_state=None, 
   solver='lbfgs', 
   max_iter=100, 
   multi_class='auto', 
   verbose=0, 
   warm_start=False, 
   n_jobs=None, 
   l1_ratio=None
)

$univariate
<class 'sklearn.feature_selection._univariate_selection.GenericUnivariateSelect'>
 signature: (
   score_func=<function f_classif at 0x000001453EDC9550>, 
   mode='percentile', 
   param=1e-05
)

$select_model
<class 'sklearn.feature_selection._from_model.SelectFromModel'>
 signature: (
   estimator, 
   threshold=None, 
   prefit=False, 
   norm_order=1, 
   max_features=None
)

$GradBoost
<class 'sklearn.ensemble._gb.GradientBoostingClassifier'>
 signature: (
   loss='deviance', 
   learning_rate=0.1, 
   n_estimators=100, 
   subsample=1.0, 
   criterion='friedman_mse', 
   min_samples_split=2, 
   min_samples_leaf=1, 
   min_weight_fraction_leaf=0.0, 
   max_depth=3, 
   min_impurity_decrease=0.0, 
   min_impurity_split=None, 
   init=None, 
   random_state=None, 
   max_features=None, 
   verbose=0, 
   max_leaf_nodes=None, 
   warm_start=False, 
   presort='deprecated', 
   validation_fraction=0.1, 
   n_iter_no_change=None, 
   tol=0.0001, 
   ccp_alpha=0.0
)

<class 'sklearn.feature_selection._univariate_selection.GenericUnivariateSelect'>
 signature: (
   score_func=<function f_classif at 0x000001453EDC9550>, 
   mode='percentile', 
   param=1e-05
)
Fitting the data split: 1 
Fitting pipeline for Lasso feature selection method
RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=100,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)
$classifier__n_estimators
[1] 100 500 900

$classifier__max_depth
[1] 10 20 30 40 50

$classifier__min_samples_split
[1]  2  5 10

$classifier__min_samples_leaf
[1] 1 2 4

$classifier__max_features
[1] "auto" "sqrt"

Fitting 5 folds for each of 10 candidates, totalling 50 fits
[Parallel(n_jobs=-1)]: Using backend LokyBackend with 32 concurrent workers.
Error in sklearn$metrics$precision_score(y_test, y_pred, average = "weighted") : 
  attempt to apply non-function`

**Comment by ybl151261** [comment]
Hello, I also encountered the same problem. May I ask if you have solved it
